src/filesystem.cpp

```diff
@@ -45,7 +45,11 @@ namespace {
 bool do_mkdir(std::string const& path, int const mode)
 {
     (void)mode; //not used on windows
+#ifdef _MSC_VER
```

Comment:
Why must this be handled two different ways within windows platforms? Is windows really that screwed up that it doesn't have a single function to do this across its versions?